homeassistant/components/overkiz/cover.py

```diff
@@ -1,40 +1,351 @@
 """Support for Overkiz covers - shutters etc."""
-from pyoverkiz.enums import OverkizCommand, UIClass
+from __future__ import annotations
 
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any, cast
+
+from pyoverkiz.enums import OverkizCommand, OverkizCommandParam, OverkizState, UIClass
+from pyoverkiz.models import Device
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    ATTR_TILT_POSITION,
+    CoverDeviceClass,
+    CoverEntity,
+    CoverEntityDescription,
+)
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import Platform
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
 from . import HomeAssistantOverkizData
 from .const import DOMAIN
-from .cover_entities.awning import Awning
-from .cover_entities.generic_cover import OverkizGenericCover
-from .cover_entities.vertical_cover import LowSpeedCover, VerticalCover
+from .entity import OverkizDescriptiveEntity
+
+
+def is_closed(device: Device) -> bool | None:
+    """Return if the cover is closed."""
+
+    if state := device.states[OverkizState.CORE_OPEN_CLOSED]:
+        return state.value == OverkizCommandParam.CLOSED
+
+    return False
+
+
+@dataclass
+class OverkizCoverDescription(CoverEntityDescription):
+    """Class to describe an Overkiz cover."""
+
+    open_command: OverkizCommand | None = None
+    close_command: OverkizCommand | None = None
+    stop_command: OverkizCommand | None = None
+    current_position_state: OverkizState | None = None
+    invert_position: bool = True
+    set_position_command: OverkizCommand | None = None
+    is_closed_fn: Callable[[Device], bool | None] | None = None
+    current_tilt_position: OverkizState | None = None
+    set_tilt_position_command: OverkizCommand | None = None
+    open_tilt_command: OverkizCommand | None = None
+    close_tilt_command: OverkizCommand | None = None
+    stop_tilt_command: OverkizCommand | None = None
+
+
+COVER_DESCRIPTIONS: list[OverkizCoverDescription] = [
+    OverkizCoverDescription(
```

Comment:
We need to set supported_features as well, otherwise this won't work like it did before.